pkg/section/module.go

```diff
@@ -0,0 +1,75 @@
+package section
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+
+	"github.com/daixiang0/gci/pkg/parse"
+	"github.com/daixiang0/gci/pkg/specificity"
+)
+
+const ModuleType = "module"
+
+type Module struct {
+	ModulePaths []string
+}
+
+func (m *Module) MatchSpecificity(spec *parse.GciImports) specificity.MatchSpecificity {
+	for _, modPath := range m.ModulePaths {
+		if strings.HasPrefix(spec.Path, modPath) {
+			return specificity.Module{}
+		}
+	}
+
+	return specificity.MisMatch{}
+}
+
+func (m *Module) String() string {
+	return ModuleType
+}
+
+func (m *Module) Type() string {
+	return ModuleType
+}
+
+// Configure configures the module section by finding the module
+// for the current path
+func (m *Module) Configure() error {
+	modPaths, err := findModules()
+	if err != nil {
+		return err
+	}
+	m.ModulePaths = modPaths
+	return nil
+}
+
+func findModules() ([]string, error) {
+	packages, err := packages.Load(
+		// find the package in the current dir and load its module
+		&packages.Config{Mode: packages.NeedModule},
+		".",
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load local modules: %v", err)
+	}
+
+	uniqueModules := make(map[string]struct{})
```

Comment:
```suggestion
	uniqueModules := make(map[string]bool)
```

Reply:
What's the reason for the preference of `bool` over `struct{}`?

I've no objection to changing it, I'm just curious. I tend to use `map[T]struct{}` in cases where only the keys are relevant

Reply:
Use bool then following codes would be reduce.

Reply:
`map[T]struct{}` is the canonical way to represent a set of `T` (strings in this case). Using a boolean instead of an empty struct provides no benefit, since you won't ever set the value to `false`.

You can answer the question "is the element `x` in the set?" by using a special case [index expression](https://go.dev/ref/spec#Index_expressions):

```go
x := "something"
var m map[string]struct{}
val, ok := m[x] //=> "", false
```

In conditions, this becomes:

```go
if _, ok := m[x]; ok {
    ...
}
```

Truly, you can achieve the same with boolean values. And not that it really matters here, but one advantage of `struct{}` is, that is **doesn't occupy memory**, whereas a `bool` requires a byte (cf. this [StackOverflow](https://stackoverflow.com/a/22770744/766783) answer):

```go
var s struct{}
fmt.Println(unsafe.Sizeof(s)) //=> 0

var b bool
fmt.Println(unsafe.Sizeof(b)) //=> 1
```

If you're really worried about code length (not a measurement I'd concentrate on, especially not in Go), you can use a local type alias:

```go
type nop = struct{} // name loosely based on the "no operation" mnemonic
var m map[string]nop
```